torchdynamo/variables/torch.py

```diff
@@ -158,6 +158,13 @@ def call_function(
             and args[0].size is not None
         ):
             return ConstantVariable(product(args[0].size), **options)
+        elif self.value is torch._shape_as_tensor:
+            assert len(args) == 1
+            assert isinstance(args[0], TensorVariable)
+            log.warning(
```

Comment:
Don't warn here, this should either be and feel safe, or not be done at all.

Reply:
I felt like since we are rewriting user code, i thought it would be good to signal them we are doing something little hacky. Maybe I can leave a TODO saying there should be more proper fix in the future.